src/deployment/deployment-hapi-plugin.ts

```diff
@@ -19,97 +20,112 @@ import { minardUiBaseUrlInjectSymbol } from '../server/types';
+              ref: Joi.string().required(),
+              sha: Joi.string(),
+            },
+          },
+        },
+      });
+    } else {
+      server.ext('onPreResponse', this.onPreResponse.bind(this));
+      server.route({
+        method: 'GET',
+        path: '/ci/deployments/{projectId}-{deploymentId}/trace',
+        handler: {
+          async: this.getTraceRequestHandler,
+        },
+        config: {
+          bind: this,
+          validate: {
+            params: {
+              projectId: Joi.number().required(),
+              deploymentId: Joi.number().required(),
+            },
+          },
+        },
+      });
+    }
     server.ext('onRequest', this.onRequest.bind(this));
-    server.ext('onPreResponse', this.onPreResponse.bind(this));
-    server.route(this.deploymentRoutes(true));
+    server.route(this.rawDeploymentRoutes(!this.isPrivate));
+    next();
+  }
 
-    server.route([{
+  private rawDeploymentRoutes(auth: boolean) {
+    return [{
       method: 'GET',
-      path: '/ci/projects/{projectId}/{ref}/{sha}/yml',
-      handler: {
-        async: this.getGitlabYmlRequestHandler,
+      path: '/deployment-favicon',
+      handler: (_request: Hapi.Request, reply: Hapi.IReply) => {
+        reply.file('favicon.ico');
       },
+      vhost: deploymentVhost,
       config: {
-        bind: this,
         auth: false,
-        validate: {
-          params: {
-            projectId: Joi.number().required(),
-            ref: Joi.string().required(),
-            sha: Joi.string(),
-          },
-        },
       },
     }, {
       method: 'GET',
-      path: '/ci/deployments/{projectId}-{deploymentId}/trace',
-      handler: {
-        async: this.getTraceRequestHandler,
-      },
+      path: '/{param*}',
+      handler: this.directoryHandler(PREKEY),
+      vhost: deploymentVhost,
       config: {
-        bind: this,
-        auth: false,
-        validate: {
-          params: {
-            projectId: Joi.number().required(),
-            deploymentId: Joi.number().required(),
-          },
-        },
+        auth: 'customAuthorize-cookie',
+        pre: this.pre(auth, true, PREKEY),
       },
-    }]);
-    next();
-  }
-
-  public registerPrivate: HapiRegister = (server, _options, next) => {
-
-    server.ext('onRequest', this.onRequest.bind(this));
-    server.route(this.deploymentRoutes(false));
-
-    next();
+    }].map((route: any) => {
+      if (!auth) {
+        route.config = { ...(route.config || {}), auth: false };
+      }
+      return route;
+    });
   }
 
   private onRequest(request: Hapi.Request, reply: Hapi.IReply) {
-    if (isRawDeploymentHostname(request.info.hostname)) {
-      // prefix the url with /raw-deployment-handler
-      // (or /deployment-favicon) to allow hapi to
-      // internally route the request to the correct handler
-
-      if (request.url.href === '/favicon.ico') {
-        request.setUrl('/deployment-favicon');
```

Comment:
I feel it's fine that deployments served from Minard have a Minard logo in them.

I think it's better to include the fix here since we want master to be in as good shape as possible, and because we want this PR to reflect all the required changes that were made for this feature. The rebase and merge shouldn't be that bad...?